dali/kernels/imgproc/color_manipulation/color_space_conversion_impl.h

```diff
@@ -86,6 +86,55 @@ DALI_HOST_DEV DALI_FORCEINLINE uint8_t rgb_to_cr(vec<3, uint8_t> rgb) {
   return static_cast<uint8_t>(dot(coeffs, rgb) + 128.0f);
 }
 
+// Gray uses the full dynamic range of the type (e.g. 0..255)
+// while ITU-R BT.601 uses a reduced range to allow for floorroom and footroom (e.g. 16..235)
+template <typename Output, typename Input>
+DALI_HOST_DEV DALI_FORCEINLINE Output y_to_gray(Input gray_in) {
+  auto gray = detail::norm(gray_in);  // TODO(janton): optimize number of multiplications
+  return ConvertSatNorm<Output>(1.164f * (gray - 0.0625f));
+}
+
+template <>
+DALI_HOST_DEV DALI_FORCEINLINE uint8_t y_to_gray(uint8_t gray) {
+  return static_cast<uint8_t>(1.164f * (gray - 16));
+}
+
+template <typename Output, typename Input>
+DALI_HOST_DEV DALI_FORCEINLINE Output gray_to_y(Input y_in) {
+  auto y = detail::norm(y_in);  // TODO(janton): optimize number of multiplications
+  constexpr float scale = 1 / 1.164f;
+  return ConvertSatNorm<Output>(y * scale + 0.0625f);
```

Comment:
```suggestion
  constexpr float scale = 0.257f +  0.504f + 0.098f;
  return ConvertSatNorm<Output>(y * scale + 0.0625f);
```
This should be exactly that.

Reply:
Done